src/Blog_App_frontend/src/AddBlog.jsx

```diff
@@ -0,0 +1,55 @@
+// AddBlog.jsx
+
+import React from 'react';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import { useNavigate } from 'react-router-dom'; // Import useNavigate hook
+import './AddBlog.css'; // Import custom CSS file for styling
+
+// Define the validation schema using Yup
+const AddBlogSchema = Yup.object().shape({
+  title: Yup.string()
+    .max(125, 'Title must not exceed 125 characters')
+    .required('Title is required'),
+  description: Yup.string().required('Description is required'),
+});
+
+function AddBlog({ handleAddBlog }) {
+  const navigate = useNavigate(); // Initialize useNavigate hook
+
+  return (
+    <div className="add-blog-container">
+      <h2 className="page-title">Add Blog</h2>
+      <Formik
+        initialValues={{ title: '', description: '' }}
+        validationSchema={AddBlogSchema}
+        onSubmit={async (values, { setSubmitting, resetForm }) => {
+          await handleAddBlog(values); // Wait for handleAddBlog to complete
+          setSubmitting(false);
+          resetForm();
+          navigate('/blogs'); // Redirect to Blogs page
+        }}
+      >
+        {({ isSubmitting }) => (
+          <Form className="add-blog-form">
+            <div className="form-group">
+              <label htmlFor="title">Title:</label>
+              <Field type="text" name="title" className="input" />
+              <ErrorMessage name="title" component="div" className="error-message" />
+            </div>
```

Comment:
use `<Form.Control.Feedback type="invalid">` from bootstrap instead of `ErrorMessage`

Reply:
changed it